src/libraries/Microsoft.PowerFx.Core/Public/Values/FormulaValue.cs

```diff
@@ -20,6 +20,11 @@ namespace Microsoft.PowerFx.Core.Public.Values
     [DebuggerDisplay("{ToObject().ToString()} ({Type})")]
     public abstract class FormulaValue
     {
+        // We place the .New*() methods on FormulaValue for discoverability. 
+        // If we're "marshalling" a T, we need a TypeMarshallerCache
+        // Else, if we're "constructing" a Table/Record from existing FormulaValues, we don't need a marshaller.
+        // We can use C# overloading to resolve. 
```

Comment:
These rules capture the principles that drive the rest of the changes in this file.